File: DC/Python/CorrectCodes/Gotham.py
class Stack:
    def __init__(self, max_size):
        self.data = []
        self.top = -1
        self.max_size = max_size

    def is_empty(self):
        return self.top == -1

    def is_full(self):
        return self.top == self.max_size - 1

    def push(self, value):
        if self.is_full():
            print("Stack overflow")
            exit(1)  # Use 1 to indicate an error

        self.top += 1
        self.data.append(value)

    def print_stack(self):
        for item in self.data[: self.top + 1]:
            print(item, end="")
        print()


def main():
    # Define the string
    my_string = ['J', 'R', 'W', 'K', 'D', 'P']
    max_size = len(my_string)  # Set the maximum size

    my_stack = Stack(max_size)

    for char in my_string:
        my_stack.push(char)

    my_stack.print_stack()


if __name__ == "__main__":
    main()

File: DC/C/correctCodes/Gotham.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 100

struct Stack {
    char data[MAX_SIZE];
    int top;
};

void initializeStack(struct Stack* stack) {
    stack->top = -1;
}

int isEmpty(struct Stack* stack) {
    return stack->top == -1;
}

int isFull(struct Stack* stack) {
    return stack->top == MAX_SIZE - 1;
}

void push(struct Stack* stack, char value) {
    if (isFull(stack)) {
        printf("Stack overflow\n");
        exit(EXIT_FAILURE);
    }

    stack->data[++stack->top] = value;
}

void printStack(struct Stack* stack) {
    for (int i = 0; i <= stack->top; i++) {
        printf("%c", (stack->data[i])-3);
    }
    printf("\n");
}

int main() {
    // Define the string
    char myString[] = {'M', 'U', 'Z', 'N', 'G', 'S'};
    int length = sizeof(myString) / sizeof(myString[0]);

    struct Stack myStack;
    initializeStack(&myStack);

    for (int i = 0; i < length; i++) {
        push(&myStack, myString[i]);
    }

    printStack(&myStack);

    return 0;
}
